src/plugins/opensearch_dashboards_react/public/overview_page/overview_page_header/overview_page_header.test.tsx

```diff
@@ -31,95 +31,265 @@
-}));
-
-afterAll(() => jest.clearAllMocks());
+jest.mock('../../context', () => ({ useOpenSearchDashboards: jest.fn() }));
 
 const mockTitle = 'Page Title';
 const addBasePathMock = jest.fn((path: string) => (path ? path : 'path'));
 
+const mockProps = () => ({
+  addBasePath: addBasePathMock,
+  title: mockTitle,
+  branding: {},
+  logos: getLogosMock.default,
+});
+
 describe('OverviewPageHeader ', () => {
-  describe('in default mode ', () => {
-    test('render logo as custom default mode logo', () => {
-      const branding = {
-        darkMode: false,
-        mark: {
-          defaultUrl: '/defaultModeLogo',
-          darkModeUrl: '/darkModeLogo',
-        },
-      };
-
-      const component = shallowWithIntl(
-        <OverviewPageHeader addBasePath={addBasePathMock} title={mockTitle} branding={branding} />
-      );
-      expect(component).toMatchSnapshot();
-    });
+  beforeAll(() => {
+    // @ts-ignore
+    useOpenSearchDashboards.mockImplementation(() => ({
+      services: {
+        application: { capabilities: { navLinks: { management: true, dev_tools: true } } },
+        notifications: { toast: { addSuccess: jest.fn() } },
+      },
+    }));
+  });
 
-    test('render logo as original default mode opensearch mark', () => {
-      const branding = {
-        darkMode: false,
-        mark: {},
-      };
+  afterAll(() => {
+    jest.clearAllMocks();
+  });
 
-      const component = shallowWithIntl(
-        <OverviewPageHeader addBasePath={addBasePathMock} title={mockTitle} branding={branding} />
-      );
-      expect(component).toMatchSnapshot();
-    });
+  it('renders without overlap by default', () => {
+    const props = {
+      ...mockProps(),
+    };
+
+    const component = shallowWithIntl(<OverviewPageHeader {...props} />);
+
+    const header = component.find('header');
+    expect(header.hasClass('osdOverviewPageHeader--hasOverlap')).toBeFalsy();
+    expect(header.hasClass('osdOverviewPageHeader--noOverlap')).toBeTruthy();
   });
 
-  describe('in dark mode ', () => {
-    test('render logo as custom dark mode logo', () => {
-      const branding = {
-        darkMode: false,
-        mark: {
-          defaultUrl: '/defaultModeLogo',
-          darkModeUrl: '/darkModeLogo',
-        },
-      };
-
-      const component = shallowWithIntl(
-        <OverviewPageHeader addBasePath={addBasePathMock} title={mockTitle} branding={branding} />
-      );
-      expect(component).toMatchSnapshot();
-    });
+  it('renders with overlap', () => {
+    const props = {
+      ...mockProps(),
+      overlap: true,
+    };
+
+    const component = shallowWithIntl(<OverviewPageHeader {...props} />);
+
+    const header = component.find('header');
+    expect(header.hasClass('osdOverviewPageHeader--hasOverlap')).toBeTruthy();
+    expect(header.hasClass('osdOverviewPageHeader--noOverlap')).toBeFalsy();
+  });
+
+  it('renders without an icon by default', () => {
+    const props = {
+      ...mockProps(),
+    };
```

Comment:
There are a few other instances where memory is allocated for no reason

```suggestion
    const props = mockProps();
```